src/sciline/pipeline.py

```diff
@@ -200,6 +205,15 @@ def _repr_html_(self) -> str:
         nodes = ((key, data) for key, data in self.underlying_graph.nodes.items())
         return pipeline_html_repr(nodes)
 
+    def leafs(self) -> tuple[type, ...]:
```

Comment:
- Can we consider making this a free function? We should avoid bloating the interface of `Pipeline`.
- What is a leaf? Can we use some clearer terminology such as used in mathematics for DAGs, or by NetworkX? Sink nodes, for example

Reply:
> What is a leaf? Can we use some clearer terminology such as used in mathematics for DAGs, or by NetworkX? Sink nodes, for example

I think `sink_nodes` is a good name, but it's slightly "leaking" the  graph abstraction below `Pipeline`. Maybe we should have something more specific to the domain, like `final_results` or `output_keys`.  

> Can we consider making this a free function? We should avoid bloating the interface of Pipeline.

I'm not so sure about this. We could make it a free function, but what's the advantage of doing so? It's tightly coupled to the `Pipeline` class. Having it be a method makes it easier for people to find it using dot-notation access.